libmamba/include/mamba/core/subdirdata.hpp

```diff
@@ -12,18 +12,22 @@
 
 #include <nlohmann/json_fwd.hpp>
 
-#include "mamba/core/channel.hpp"
-#include "mamba/core/context.hpp"
 #include "mamba/core/download.hpp"
 #include "mamba/core/error_handling.hpp"
 #include "mamba/core/package_cache.hpp"
-#include "mamba/core/pool.hpp"
 #include "mamba/core/repo.hpp"
 #include "mamba/core/util.hpp"
 #include "mamba/fs/filesystem.hpp"
 
 namespace mamba
 {
+    namespace specs
+    {
+        class Channel;
```

Comment:
Why moving this class to the `specs` folder (and inside the specs namespace)? I agree with the overall architecture, but for me Channel is the result of resolving the spec, and therefore is a core component. O do you plan to have core depending on `specs` and you want to avoid circular dependencies between layers?

Reply:
Maybe the `specs` namespace is not adequately named. It is intended to hold all abstractions that could in theory be exchanged with Conda. All the abstraction are *descriptive*: they are functional (no side effects), and hold only data. A protocol in a sense. It has `VersionSpec` and `Version`, `ChannelSpec`, `Platform`...

`ChannelContext` is a glue/machinery component and does not belong there.